src/QMCWaveFunctions/LCAO/MultiFunctorAdapter.h

```diff
@@ -57,6 +59,31 @@ struct MultiFunctorAdapter
       u[i] = Rnl[i]->f(r);
   }
 
+  inline void batched_evaluate(OffloadArray2D& r, OffloadArray3D& u, RealType Rmax) const
```

Comment:
Missing documentation. Cannot understand what are r and u. Need to document u r and u and mention the expected data locality about whether they are up-to-date on host or device or both.

Reply:
I added documentation.
r is [n_elec, n_pbc]; u is [n_elec, n_pbc, n_Rnl]

n_elec and n_pbc aren't really important on their own, and it wouldn't change anything if they were flattened to one dimension (this is the same in MultiQuinticSpline1D and Soa{Cartesian,Spherical}Tensor), but outside of this function the n_elec, n_pbc need to be treated separately.

This could be written in a more general way where r is 1D and u is 2D, but seems like too much work for now.  If I need to add one for 1D r and 2D u, I can have them both call some _impl function where I just pass array dims and pointers to the start of u/r.